### ISSUE TITLE
build fails on macOS Sierra and homebrew LLVM

### ISSUE THREAD

**Issue by lazzarello** [fundamental/stoat#32]
macOS uses Clang as the default compiler for the platform but omits LLVM tools required to build this project. The homebrew package manager provides LLVM with an option to use environment variables for programs like stoat which need to link to LLVM libraries. Building fails with the following output. My system details follow.

```
[lazzarello@strop build (master)]$ cmake ..
-- The C compiler identification is AppleClang 8.0.0.8000042
-- The CXX compiler identification is AppleClang 8.0.0.8000042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found LLVM: -lLLVMLTO -lLLVMPasses -lLLVMObjCARCOpts -lLLVMSymbolize -lLLVMDebugInfoPDB -lLLVMDebugInfoDWARF -lLLVMMIRParser -lLLVMCoverage -lLLVMTableGen -lLLVMDlltoolDriver -lLLVMOrcJIT -lLLVMXCoreDisassembler -lLLVMXCoreCodeGen -lLLVMXCoreDesc -lLLVMXCoreInfo -lLLVMXCoreAsmPrinter -lLLVMSystemZDisassembler -lLLVMSystemZCodeGen -lLLVMSystemZAsmParser -lLLVMSystemZDesc -lLLVMSystemZInfo -lLLVMSystemZAsmPrinter -lLLVMSparcDisassembler -lLLVMSparcCodeGen -lLLVMSparcAsmParser -lLLVMSparcDesc -lLLVMSparcInfo -lLLVMSparcAsmPrinter -lLLVMPowerPCDisassembler -lLLVMPowerPCCodeGen -lLLVMPowerPCAsmParser -lLLVMPowerPCDesc -lLLVMPowerPCInfo -lLLVMPowerPCAsmPrinter -lLLVMNVPTXCodeGen -lLLVMNVPTXDesc -lLLVMNVPTXInfo -lLLVMNVPTXAsmPrinter -lLLVMMSP430CodeGen -lLLVMMSP430Desc -lLLVMMSP430Info -lLLVMMSP430AsmPrinter -lLLVMMipsDisassembler -lLLVMMipsCodeGen -lLLVMMipsAsmParser -lLLVMMipsDesc -lLLVMMipsInfo -lLLVMMipsAsmPrinter -lLLVMLanaiDisassembler -lLLVMLanaiCodeGen -lLLVMLanaiAsmParser -lLLVMLanaiDesc -lLLVMLanaiAsmPrinter -lLLVMLanaiInfo -lLLVMHexagonDisassembler -lLLVMHexagonCodeGen -lLLVMHexagonAsmParser -lLLVMHexagonDesc -lLLVMHexagonInfo -lLLVMBPFDisassembler -lLLVMBPFCodeGen -lLLVMBPFDesc -lLLVMBPFInfo -lLLVMBPFAsmPrinter -lLLVMARMDisassembler -lLLVMARMCodeGen -lLLVMARMAsmParser -lLLVMARMDesc -lLLVMARMInfo -lLLVMARMAsmPrinter -lLLVMAMDGPUDisassembler -lLLVMAMDGPUCodeGen -lLLVMAMDGPUAsmParser -lLLVMAMDGPUDesc -lLLVMAMDGPUInfo -lLLVMAMDGPUAsmPrinter -lLLVMAMDGPUUtils -lLLVMAArch64Disassembler -lLLVMAArch64CodeGen -lLLVMAArch64AsmParser -lLLVMAArch64Desc -lLLVMAArch64Info -lLLVMAArch64AsmPrinter -lLLVMAArch64Utils -lLLVMObjectYAML -lLLVMLibDriver -lLLVMOption -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMGlobalISel -lLLVMSelectionDAG -lLLVMAsmPrinter -lLLVMDebugInfoCodeView -lLLVMDebugInfoMSF -lLLVMX86Desc -lLLVMMCDisassembler -lLLVMX86Info -lLLVMX86AsmPrinter -lLLVMX86Utils -lLLVMMCJIT -lLLVMLineEditor -lLLVMInterpreter -lLLVMExecutionEngine -lLLVMRuntimeDyld -lLLVMCodeGen -lLLVMTarget -lLLVMCoroutines -lLLVMipo -lLLVMInstrumentation -lLLVMVectorize -lLLVMScalarOpts -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMObject -lLLVMMCParser -lLLVMMC -lLLVMBitReader -lLLVMCore -lLLVMBinaryFormat -lLLVMSupport -lLLVMDemangle;-L/usr/local/Cellar/llvm/5.0.0/lib -Wl,-search_paths_first -Wl,-headerpad_max_install_names (found version "5.0.0") 
statusBuild Flags ' -I/usr/local/Cellar/llvm/5.0.0/include   -DLLVM_BUILD_GLOBAL_ISEL -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-rtti -std=c++11'
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   stoat

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/lazzarello/src/stoat/build
[lazzarello@strop build (master)]$ make
Scanning dependencies of target stoat
[ 50%] Building CXX object CMakeFiles/stoat.dir/src/llvm-passes.cpp.o
/Users/lazzarello/src/stoat/src/llvm-passes.cpp:536:45: error: no member named 'getArgumentList' in
      'llvm::Function'
                std::string alias_type = Fn.getArgumentList().front().getType()->getPointerElementTyp...
                                         ~~ ^
1 error generated.
make[2]: *** [CMakeFiles/stoat.dir/src/llvm-passes.cpp.o] Error 1
make[1]: *** [CMakeFiles/stoat.dir/all] Error 2
make: *** [all] Error 2
```

## system info

```
[lazzarello@strop stoat (master)]$ llvm-config --version
5.0.0
[lazzarello@strop stoat (master)]$ clang --version
clang version 5.0.0 (tags/RELEASE_500/final)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
[lazzarello@strop stoat (master)]$ cmake --version
cmake version 3.10.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

**Comment by fundamental** [comment]
My distro is currently on LLVM 3.8.0, so I can't fully test the fix, but based upon https://github.com/llvm-mirror/llvm/blob/master/include/llvm/IR/Function.h the problem should now be solved on your system.

**Comment by lazzarello** [comment]
Hmmm, well, it compiles with the new function but the linker throws an error. My intuition says I could figure this out but this is literally my first time looking at LLVM internals...perhaps you might have quicker intuition?

```
strop:build lazzarello$ make
Scanning dependencies of target stoat
[ 50%] Building CXX object CMakeFiles/stoat.dir/src/llvm-passes.cpp.o
[100%] Linking CXX shared library libstoat.dylib
Undefined symbols for architecture x86_64:
  "llvm::ModulePass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)", referenced from:
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
  "llvm::ModulePass::~ModulePass()", referenced from:
      (anonymous namespace)::ExtractAnnotations::~ExtractAnnotations() in llvm-passes.cpp.o
      (anonymous namespace)::ExtractVtables::~ExtractVtables() in llvm-passes.cpp.o
  "llvm::raw_ostream::flush_nonempty()", referenced from:
      llvm::raw_ostream::flush() in llvm-passes.cpp.o
  "llvm::raw_ostream::write(char const*, unsigned long)", referenced from:
      llvm::raw_ostream::operator<<(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in llvm-passes.cpp.o
      llvm::raw_ostream::operator<<(llvm::StringRef) in llvm-passes.cpp.o
  "llvm::raw_ostream::write(unsigned char)", referenced from:
      llvm::raw_ostream::operator<<(char) in llvm-passes.cpp.o
  "llvm::FunctionPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::PassRegistry::registerPass(llvm::PassInfo const&, bool)", referenced from:
      llvm::RegisterPass<(anonymous namespace)::ExtractCallGraph>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractAnnotations>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractClassHierarchy>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractVtables>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractRtosc>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
  "llvm::PassRegistry::getPassRegistry()", referenced from:
      llvm::RegisterPass<(anonymous namespace)::ExtractCallGraph>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractAnnotations>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractClassHierarchy>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractVtables>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
      llvm::RegisterPass<(anonymous namespace)::ExtractRtosc>::RegisterPass(llvm::StringRef, llvm::StringRef, bool, bool) in llvm-passes.cpp.o
  "llvm::raw_string_ostream::~raw_string_ostream()", referenced from:
      (anonymous namespace)::typeToString(llvm::Type*) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractClassHierarchy::extractSuperClass(llvm::Instruction*, llvm::BitCastInst*, bool&) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractRtosc::isCallback(llvm::GetElementPtrInst*) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractRtosc::isName(llvm::GetElementPtrInst*) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractRtosc::isMeta(llvm::GetElementPtrInst*) in llvm-passes.cpp.o
  "llvm::DisableABIBreakingChecks", referenced from:
      llvm::VerifyDisableABIBreakingChecks in llvm-passes.cpp.o
  "llvm::Pass::releaseMemory()", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::dumpPassStructure(unsigned int)", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::getAsImmutablePass()", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::getAsPMDataManager()", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::preparePassManager(llvm::PMStack&)", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::getAdjustedAnalysisPointer(void const*)", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::~Pass()", referenced from:
      llvm::FunctionPass::~FunctionPass() in llvm-passes.cpp.o
  "llvm::errs()", referenced from:
      (anonymous namespace)::ExtractRtosc::runOnFunction(llvm::Function&) in llvm-passes.cpp.o
      (anonymous namespace)::escapeOutput(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in llvm-passes.cpp.o
  "llvm::ModulePass::createPrinterPass(llvm::raw_ostream&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
  "llvm::ModulePass::getPotentialPassManagerType() const", referenced from:
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
  "llvm::StructType::getName() const", referenced from:
      llvm::Type::getStructName() const in llvm-passes.cpp.o
  "llvm::FunctionPass::createPrinterPass(llvm::raw_ostream&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::FunctionPass::getPotentialPassManagerType() const", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::ConstantDataSequential::getRawDataValues() const", referenced from:
      llvm::ConstantDataSequential::getAsString() const in llvm-passes.cpp.o
      (anonymous namespace)::ExtractRtosc::runOnFunction(llvm::Function&) in llvm-passes.cpp.o
  "llvm::ConstantDataSequential::isString(unsigned int) const", referenced from:
      llvm::ConstantDataSequential::getAsString() const in llvm-passes.cpp.o
  "llvm::Pass::getPassName() const", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::verifyAnalysis() const", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Pass::print(llvm::raw_ostream&, llvm::Module const*) const", referenced from:
      vtable for (anonymous namespace)::ExtractCallGraph in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractAnnotations in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractClassHierarchy in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractVtables in llvm-passes.cpp.o
      vtable for (anonymous namespace)::ExtractRtosc in llvm-passes.cpp.o
  "llvm::Type::print(llvm::raw_ostream&, bool, bool) const", referenced from:
      (anonymous namespace)::typeToString(llvm::Type*) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractClassHierarchy::extractSuperClass(llvm::Instruction*, llvm::BitCastInst*, bool&) in llvm-passes.cpp.o
  "llvm::APInt::countLeadingOnesSlowCase() const", referenced from:
      llvm::APInt::countLeadingOnes() const in llvm-passes.cpp.o
  "llvm::APInt::countLeadingZerosSlowCase() const", referenced from:
      llvm::APInt::countLeadingZeros() const in llvm-passes.cpp.o
  "llvm::Value::getName() const", referenced from:
      (anonymous namespace)::ExtractCallGraph::runOnFunction(llvm::Function&) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractCallGraph::runOnCallInst(llvm::CallInst*, llvm::Function&) in llvm-passes.cpp.o
      (anonymous namespace)::isCatchCall(llvm::LoadInst*) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractAnnotations::handleNested(llvm::Value*) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractClassHierarchy::runOnFunction(llvm::Function&) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractVtables::runOnModule(llvm::Module&) in llvm-passes.cpp.o
      (anonymous namespace)::ExtractVtables::handleVtable(char const*, llvm::ConstantArray*) in llvm-passes.cpp.o
      ...
  "llvm::Module::getGlobalVariable(llvm::StringRef, bool) const", referenced from:
      llvm::Module::getNamedGlobal(llvm::StringRef) const in llvm-passes.cpp.o
  "llvm::Constant::getAggregateElement(unsigned int) const", referenced from:
      (anonymous namespace)::ExtractVtables::runOnModule(llvm::Module&) in llvm-passes.cpp.o
  "llvm::Function::BuildLazyArguments() const", referenced from:
      llvm::Function::CheckLazyArguments() const in llvm-passes.cpp.o
  "vtable for llvm::ModulePass", referenced from:
      llvm::ModulePass::ModulePass(char&) in llvm-passes.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for llvm::raw_ostream", referenced from:
      llvm::raw_ostream::raw_ostream(bool) in llvm-passes.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for llvm::FunctionPass", referenced from:
      llvm::FunctionPass::FunctionPass(char&) in llvm-passes.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for llvm::raw_string_ostream", referenced from:
      llvm::raw_string_ostream::raw_string_ostream(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in llvm-passes.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for llvm::Pass", referenced from:
      llvm::Pass::Pass(llvm::PassKind, char&) in llvm-passes.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libstoat.dylib] Error 1
make[1]: *** [CMakeFiles/stoat.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by fundamental** [comment]
Based upon stackoverflow it looks like the answer is the "-undefined dynamic_lookup" linker flag https://stackoverflow.com/questions/3695234/what-is-the-deal-with-undefined-symbols-in-a-shared-library-or-dylib . You should be able to test if this is the solution by messing with "ccmake ." via the CMAKE_SHARED_LINKER_FLAGS config variable and then confirming the flag is used via "make VERBOSE=1". Let me know if this resolves the problem.

**Comment by lazzarello** [comment]
Wow! Yes, that works. Thanks for your help!

Should I make a PR updating the README with this thread? Or perhaps a wiki page?

**Comment by fundamental** [comment]
This isn't a documentation issue per-say, but a bug in the cmake script on the mac os platform. basically there needs to be a 

    if(OPERATING_SYSTEM STREQUALS "MAC");
        SET(CMAKE_SHARED_LINKER_FLAGS "-undefined dynamic_lookup");
    endif()

With the valid if() condition. If you can find the right platform detection routine and add that, then it would make for a good pull request. Otherwise when I have a chance I should be able to look it up and add the platform specific linker flag on a later date.

**Comment by fundamental** [comment]
fixed via pr #33
